reqh/reqh.c

```diff
@@ -213,7 +213,7 @@ m0_reqh_mdpool_service_index_to_session(const struct m0_reqh *reqh,
 		pd_sdev_idx;
 	ctx = md_pv->pv_pc->pc_dev2svc[idx].pds_ctx;
 	M0_ASSERT(ctx != NULL);
-	M0_ASSERT(ctx->sc_type == M0_CST_IOS);
+	M0_ASSERT(ctx->sc_type == M0_CST_CAS);
```

Comment:
This should be IOS only,  MDPOOL associated with IOS needs to have dummy devices or first device in each m0d to use mdcobs associated with the m0d's.

They are not related to CAS service.

Reply:
Suppose if CAS service is part of separate m0d where m0d isn ot present, this won't work.

Reply:
Also even if this change is there, issue is still coming up,

 # ./build-deploy/cortx-motr/utils/m0cp -l inet:tcp:10.230.250.75@5001 -H inet:tcp:10.230.250.75@2001 -p  0x7000000000000001:0x4f -P 0x7200000000000001:0x29  -o 12:39 -s
 1m -c 1 -L 9 /tmp/128M
motr[126663]:  4cf0 ALWAYS  [client_init.c:468:client_net_init]  trasnport ep:inet:tcp:10.230.250.75@5001
motr[126663]:  4650  FATAL  [lib/assert.c:50:m0_panic]  panic: (cas_svc->sc_type == M0_CST_CAS) at dix_cas_rops_send() (dix/req.c:1749)  [git: 2.0.0-527-112-g409f711-dirty] /root/m0trace.126663
Motr pan

Reply:
@madhavemuri, so, we have 2 devices then,
1. CAS device, is that the meta_data device mentioned in the CDF?
2. IOS md device, there's no other mechanism to specify IOS metadata device and one device cannot be associated with multiple service types.
>MDPOOL associated with IOS needs to have dummy devices or first device in each m0d to use mdcobs associated with the m0d's.

- dummy device - how will it be used?
- first device of ioservice - that means it is same as the data device and will share space with data. Is that what we want?, will the first device assumption always hold true? or do we want an explicit mechanism to mention the CAS device and IOS md device separately?

> Also even if this change is there, issue is still coming up,

@madhavemuri, This means a wrong CAS device is being accessed. It seems, the problem is in the clear differentiation and assignment of metadata devices, which device must be used for CAS and which for IOS metadata.